src/plugins/js-transpilers/Babel7Plugin.ts

```diff
@@ -0,0 +1,170 @@
+import * as fs from "fs";
+import * as path from "path";
+import { File } from "../../core/File";
+import { string2RegExp } from "../../Utils";
+import { WorkFlowContext } from "../../core/WorkflowContext";
+import { Plugin } from "../../core/WorkflowContext";
+
+let babel7Core;
+
+/**
+ * @export
+ * @class FuseBoxBabel7Plugin
+ * @implements {Plugin}
+ */
+export class Babel7PluginClass implements Plugin {
+	/**
+	 * We can add tsx and ts here as well
+	 * Because Babel won't capture it just being a Plugin
+	 * Typescript files are handled before any external plugin is executed
+	 */
+	public extensions: Array<string> = [".jsx"];
+	public test: RegExp = /\.(j|t)s(x)?$/;
+	public context: WorkFlowContext;
+	private limit2project: boolean = true;
+
+	private config?: any = {};
+	private configPrinted = false;
+	private configLoaded = false;
+
+	constructor(opts: any = { ast: true }) {
+		// if it is an object containing only a babel config
+		if (
+			opts.config === undefined &&
+			opts.test === undefined &&
+			opts.limit2project === undefined &&
+			opts.extensions === undefined &&
+			Object.keys(opts).length
+		) {
+			this.config = opts;
+			return;
+		}
+
+		if (opts.config) {
+			this.config = Object.assign({ ast: true }, opts.config);
+		}
+		if (opts.extensions !== undefined) {
+			this.extensions = opts.extensions;
+			if (opts.test === undefined) {
+				this.test = string2RegExp(opts.extensions.join("|"));
+			}
+		}
+		if (opts.test !== undefined) {
+			this.test = opts.test;
+		}
+		if (opts.limit2project !== undefined) {
+			this.limit2project = opts.limit2project;
+		}
+	}
+
+	/**
+	 * @see this.init
+	 */
+	private handleBabelRc() {
+		if (this.configLoaded) return;
+
+		let babelRcConfig;
+		let babelRcPath = path.join(this.context.appRoot, `.babelrc`);
+
+		if (fs.existsSync(babelRcPath)) {
+			babelRcConfig = fs.readFileSync(babelRcPath).toString();
+
+			if (babelRcConfig) {
+				babelRcConfig = Object.assign({}, JSON.parse(babelRcConfig), this.config);
+			}
+		}
+
+		if (babelRcConfig) {
+			this.config = babelRcConfig;
+		}
+
+		this.configLoaded = true;
+	}
+
+	/**
+	 * @param {WorkFlowContext} context
+	 */
+	public init(context: WorkFlowContext) {
+		this.context = context;
+		if (Array.isArray(this.extensions)) {
+			this.extensions.forEach(ext => context.allowExtension(ext));
+		}
+		this.handleBabelRc();
+	}
+
+	/**
+	 * @param {File} file
+	 */
+	public transform(file: File) {
+		file.wasTranspiled = true;
+		if (!babel7Core) {
+			babel7Core = require("@babel/core");
```

Comment:
Shall we add exception here? We should say that it's required to install `@babel/core`